### PR TITLE
Add docs and doc building action

### LINKED ISSUE #77: Add documentation compilation and integration testing

### PR DESCRIPTION
**Pull request by craffel**
Fixes #77

### CODE REVIEW

**Comment by craffel** [comment]
I fixed the doc building action. It now runs successfully, so we can now hopefully see if anyone writes a docstring with a syntax error automatically.

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - uses: actions/checkout@v2
+    - name: Set Up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.8"
+    - name: Install Dependencies and Package
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install .[all,docs]
+        python -m pip install sphinx
```
Comment (blester125) [inline .github/workflows/build_docs.yml:26]: redundant with the new `docs` target?

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+# -- Project information -----------------------------------------------------
+
+project = "git-theta"
+copyright = "2023, r-three"
+author = "r-three"
+
+# The full version, including alpha/beta/rc tags
+release = "0"
```
Comment (blester125) [inline docs/source/conf.py:26]: When we actually use these docs I think we should be able to import git_theta here and have the release number be the actually version number.

**Review by blester125 — APPROVED** [review]

File: .github/workflows/build_docs.yml (hunk 1 above)
Comment (craffel) [reply .github/workflows/build_docs.yml]: Done.

File: docs/source/conf.py (hunk 2 above)
Comment (craffel) [reply docs/source/conf.py]: Done.

**Pull request merged**